connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorMaker.java

```diff
@@ -257,10 +285,22 @@ private void addHerder(SourceAndTarget sourceAndTarget) {
         // tracking the various shared admin objects in this class.
         Herder herder = new DistributedHerder(distributedConfig, time, worker,
                 kafkaClusterId, statusBackingStore, configBackingStore,
-                advertisedUrl, CLIENT_CONFIG_OVERRIDE_POLICY, sharedAdmin);
+                advertisedUrl, CLIENT_CONFIG_OVERRIDE_POLICY,
+                restNamespace, sharedAdmin);
         herders.put(sourceAndTarget, herder);
     }
 
+    private static String encodePath(String rawPath) throws UnsupportedEncodingException {
+        return URLEncoder.encode(rawPath, StandardCharsets.UTF_8.name())
+                // Java's out-of-the-box URL encoder encodes spaces (' ') as pluses ('+'),
+                // and pluses as '%2B'
+                // But Jetty doesn't decode pluses at all and leaves them as-are in decoded
+                // URLs
+                // So to get around that, we replace pluses in the encoded URL here with '%20',
+                // which is the encoding that Jetty expects for spaces
+                .replaceAll("\\+", "%20");
```

Comment:
I went through and tested this to prove it to myself, because I didn't believe the comment.
```suggestion
                // Jetty will reverse this transformation when evaluating the path parameters
                // and will return decoded strings with all special characters as they were.
                .replaceAll("\\+", "%20");
```